### ISSUE TITLE
Copying data

### ISSUE THREAD

**Issue by Danwhy** [LDMW/app#487]
Once we have determined the structure of the data (#486), we need to ensure we can easily copy the data we currently have on the live site to the new wagtail instance. How much the structure of the data changes will determine the method we use for this, but hopefully, with little change to the resources, we should be able to load the majority of the data straight in.

**Comment by reddog** [comment]
@Danwhy Can you expand a little on why we are creating a new Wagtail instance (requiring the data to be copied over).  Why can't the existing Wagtail instance be used and the various templates/JS/CSS/Python be modified around it?

**Comment by nelsonic** [comment]
@Danwhy as discussed on the call last week, we need to open an issue to determining `if` we _need_ a `new` Wagtail instance. (_on the surface, I don't see the advantage_)

**Comment by shouston3** [comment]
The reason for this was that @Danwhy and I are planning to change how the wagtail data is currently stored to make page nesting make more sense and tagging better organised.

From #488 we'll now work from this set of issues: https://github.com/LDMW/cms/issues and work from our current cms code

**Comment by Danwhy** [comment]
Our reasoning for wanting to use a new instance of Wagtail was because the current instance is set up in such a way that it was easy for us to pull data through to Phoenix. This doesn't appear to be the same way we'd structure the app if we were just using Wagtail, which works best with the models/data structured in a certain way.

We could keep the current cms and build on top of it, but it would require changing the models in such a way that we would have to create many migrations deleting/renaming current classes, which we know from experience is quite difficult and messy. I also don't see many advantages to keeping the current cms, beyond the data, which should hopefully be trivial to migrate, providing we keep the structure of the homepage, and the individual resources (which is the current plan). The current cms has minimal code, as most of what we were doing has taken place in this repo.

We were planning to branch from this repo, rather than the cms one, as this would allow us to keep all of the history of our issues and pull requests, which have predominantly been kept in here.

**Comment by Danwhy** [comment]
Maybe we should spend an hour or so spiking the proposed changes to the data, as it could turn out to not be as difficult as we think

**Comment by Danwhy** [comment]
After looking into the wagtail documentation a little more with specific regard to moving/restructuring pages, we've found that moving pages around is fairly simple, so we're going to go with branching off of the current cms repo.

We'll still need to set up a new wagtail instance while we get this ready, as we want to leave both the live and staging site intact until we're ready to switch over, but we can load the live data straight in as the initial code and migrations will all be the same.

**Comment by nelsonic** [comment]
@Danwhy nice. (_good detective skills!_)
If you weren't _already_ going to take a DB snapshot `before` doing the migration, 
please do so and upload a _zipped_ SQL dump file _somewhere_ 😉

**Comment by Danwhy** [comment]
:+1: Heroku backs up a SQL file daily, and the last one was five hours age. We also pulled down a json dump of the data, which we used to add the data to the new cms branch.

**Comment by reddog** [comment]
Sounds like some really useful research in terms of page structure.  Doing things *the Wagtail way* might mean an easier to understand / support / further develop solution.

Are there any time/effort implications in moving from the structure that was Phoenix-friendly (directly extract data from the database) to *the Wagtail way*?  Seems like there's not from what was said above.

**Comment by Danwhy** [comment]
No, I think it was an expected change, so the majority of the time cost has already been spent in looking into how to restructure wagtail. I think that now that we know how to do this, it should make it relatively simple to make the changes we need, and even change them again in the future if necessary.
